service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageQueries.kt

```diff
@@ -220,9 +225,17 @@ SELECT
     msg.read_at,
     msg.recipient_id,
     msg.recipient_name,
-    msg.recipient_account_type
+    msg.recipient_account_type,
+    (
+        SELECT coalesce(jsonb_agg(json_build_object(
+           'id', att.id,
+           'name', att.name,
+           'contentType', att.content_type
+        )) FILTER ( WHERE att.id IS NOT NULL ), '[]') 
```

Comment:
What is the `FILTER` clause needed for? 🤔

Reply:
I think it otherwise produced an object with null values

Reply:
Ach, the coalesce default should probably be `'[]'::jsonb`.

Reply:
huh, that works